stetho-repl-js/src/main/java/com/facebook/stetho/inspector/console/js/JavaScriptRuntimeRepl.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.console.js;
+
+import com.facebook.stetho.common.LogUtil;
+import com.facebook.stetho.inspector.console.RuntimeRepl;
+import org.mozilla.javascript.Context;
+import org.mozilla.javascript.ScriptableObject;
+
+class JavaScriptRuntimeRepl implements RuntimeRepl {
+
+  private final ScriptableObject mJsScope;
+
+  JavaScriptRuntimeRepl(ScriptableObject scope) {
+    mJsScope = scope;
+  }
+
+  @Override
+  public Object evaluate(String expression) throws Throwable {
+      LogUtil.i("js) eval expression: %s", expression);
+
+      Object result;
+      final Context jsContext = enterJsContext();
+      try {
+        result = jsContext.evaluateString(mJsScope, expression, "chrome", 1, null);
+      } finally {
+        Context.exit();
+      }
+
+      LogUtil.i("js) Result: %s; type: %s", result, result == null ? null : result.getClass().getName());
+      return Context.jsToJava(result, Object.class);
+  }
+
+  /**
+   * Setups a proper javascript context so that it can run javascript code properly under android.
+   * For android we need to disable bytecode generation since the android vms don't understand JVM bytecode.
+   * @return a proper javascript context
+   */
+  static Context enterJsContext() {
+    LogUtil.i("js) init JS context");
+    final Context jsContext = Context.enter();
+
+    // If we cause the context to throw a runtime exception from this point
+    // we need to make sure that exit the context.
+    try {
+      jsContext.setLanguageVersion(Context.VERSION_1_8);
+
+      // We can't let Rhino to optimize the JS and to use a JIT because it would generate JVM bytecode
+      // and android runs on DEX bytecode. Instead we need to go in interpreted mode.
+      jsContext.setOptimizationLevel(-1);
+    }
+    catch (RuntimeException e) {
+      Context.exit();
```

Comment:
I reviewed the code and i do believe that i was missing a ```throw e``` statement in there. It's now added.